src/App/Logger/LogFile.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CloudPatches\App\Logger;
+
+use Magento\CloudPatches\Filesystem\FileList;
+use Magento\CloudPatches\Filesystem\Filesystem;
+use Magento\CloudPatches\Filesystem\FileSystemException;
+
+/**
+ * Uses for copying log file from build phase if needed.
+ */
+class LogFile
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var FileList
+     */
+    private $fileList;
+
+    /**
+     * @param Filesystem $filesystem
+     * @param FileList $fileList
+     *
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        FileList $fileList
+    ) {
+        $this->filesystem = $filesystem;
+        $this->fileList = $fileList;
+    }
+
+    /**
+     * Checks and copies log file from build phase if needed.
+     *
+     * Creates a log directory if it doesn't exist.
+     *
+     * @return bool
+     */
+    public function prepare(): bool
+    {
+        $deployLogPath = $this->fileList->getPatchLog();
+        $buildPhaseLogPath = $this->fileList->getInitPatchLog();
+
+        $deployLogDir = $this->filesystem->getDirectory($deployLogPath);
+        $this->filesystem->createDirectory($deployLogDir);
+
+        try {
+            if ($this->isNeedToCopyBuildErrorLogFile($deployLogPath, $buildPhaseLogPath)) {
+                $this->filesystem->copy($buildPhaseLogPath, $deployLogPath);
+            }
+        } catch (FileSystemException $e) {
+            // Do nothing
+        }
+
+        return $this->filesystem->isWritable($deployLogDir);
+    }
+
+    /**
+     * Checks if log file from build phase should be copied
+     *
+     * @param string $deployLogPath
+     * @param string $buildPhaseLogPath
+     *
+     * @return bool
+     * @throws FileSystemException
+     */
+    private function isNeedToCopyBuildErrorLogFile(string $deployLogPath, string $buildPhaseLogPath): bool
```

Comment:
I don't see how this code will work, `cloud-patches` application is never created on the deploy phase only on the build.
Logic about copying `patch.log` file from `init` directory should be moved to `ece-tools` deploy phase.

Reply:
Currently, it copies patch.log from init directory during instantiation \Magento\CloudPatches\App\Logger
Where would the proper place for this logic in ece-tools? Here https://github.com/magento/ece-tools/blob/develop/src/App/Logger.php#L62 ?

Reply:
Yes, but `cloud-patches` application is not creating on the deploy phase, so the file won't be copied. 
https://github.com/magento/ece-tools/blob/develop/src/App/Logger.php class is not a good place for this logic because it's only about `cloud.log` file.
It's better to add this in some step at the beginning of deploy phase or create a new one.
For example you can add new sub-step in PreDeploy step.
https://github.com/magento/ece-tools/blob/develop/scenario/deploy.xml#L8
```
<item name="restore-files" xsi:type="object" priority="25">Magento\MagentoCloud\Step\Deploy\PreDeploy\RestoreFiles</item>
```

Reply:
Done
Logic about copying patch.log file from init directory moved to ece-tools, PR https://github.com/magento/ece-tools/pull/750